src/hotspot/share/utilities/globalCounter.hpp

```diff
@@ -63,9 +64,14 @@ class GlobalCounter : public AllStatic {
   class CounterThreadCheck;
 
  public:
+
+
   // The type of the critical section context passed from
   // critical_section_begin() to critical_section_end().
-  typedef uintx CSContext;
+  enum class CSContext : uintx {}; // [COUNTER_ACTIVE, COUNTER_INCREMENT)
```

Comment:
This enum declaration instead of the typedef is all that's needed.  The range comment is incorrect.

Reply:
So, all this change needs is:  "enum class CSContext: uintx {};"   ?

Reply:
Yes.  The RFE was filed as a followup to JDK-8212827, where I did everything needed to make that work, but couldn't take that final step because we weren't supporting C++11/14 yet.